### PR TITLE
DBZ-2604 Use default db charset as default table charset

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2604

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/CreateTableParserListener.java [hunk 1]
```diff
@@ -55,7 +55,8 @@ public void exitColumnCreateTable(MySqlParser.ColumnCreateTableContext ctx) {
         parser.runIfNotNull(() -> {
             // Make sure that the table's character set has been set ...
             if (!tableEditor.hasDefaultCharsetName()) {
-                tableEditor.setDefaultCharsetName(parser.currentDatabaseCharset());
+                final String defaultDatabaseCharset = parser.charsetNameForDatabase().get(tableEditor.tableId().catalog());
+                tableEditor.setDefaultCharsetName(defaultDatabaseCharset != null ? defaultDatabaseCharset : parser.currentDatabaseCharset());
```
Comment (victorxiang30) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/CreateTableParserListener.java:59]: hi @jpechane May I know why previous parser.currentDatabaseCharset() does not return database encoding?

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/CreateTableParserListener.java:59]: It reads `character_set_database` db variable which is set to the database to which Debezium JDBC client is connected, not to table/database DDL which is now parsing and processing.

Comment (victorxiang30) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/CreateTableParserListener.java]: I see. Does this mean db variable will override table/database DDL which is now parsing? How about the cases when column level is `UTF8mb4` but db variable is latin2. In this case, the column will be misintepreted as latin2 but is supposed to be encoded as `UTF8mb4`

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/CreateTableParserListener.java]: No the logic is
 - if column has its own charset then use it
 - if table has default charset then use it
 - if database to which the table belongs has a charset value then use it

**Comment by gunnarmorling** [comment]
LGTM, thx. Unrelated test failure tracked via https://issues.redhat.com/browse/DBZ-2625.

**Pull request merged**